src/prefect_server/graphql/agents.py

```diff
@@ -0,0 +1,61 @@
+from typing import Any, List
+
+from graphql import GraphQLResolveInfo
+
+from prefect_server import api
+from prefect_server.utilities import context
+from prefect_server.utilities.graphql import mutation
+
+
+@mutation.field("register_agent_instance")
+async def resolve_register_agent_instance(
+    obj: Any, info: GraphQLResolveInfo, input: dict
+) -> dict:
+    agent_id = await api.agents.register_agent_instance(
+        tenant_id=input.get("tenant_id"),
+        labels=input.get("labels", []) or [],
```

Comment:
These are required fields in the GQL schema, so you can rely on them having expected types
```suggestion
        labels=input["labels"],
```